### PR TITLE
Host Data From External Systems

### LINKED ISSUE #61: Design How Host Data Can Come From Other Systems
In support of allowing other systems to provide host data in a brown field environment
I would like ``Host`` to be abstracted in such a way that it may be from N number of horizontal systems
so that I do not have to have multiple copies of host inventories.

### PR DESCRIPTION
**Pull request by ashcrow**
See #61

### CODE REVIEW

**Comment by ashcrow** [comment]
[[multi-host-mgr-2] [spike] Design host abstraction](https://trello.com/c/TvgomncK/579-multi-host-mgr-2-spike-design-host-abstraction)

**Comment by ashcrow** [comment]
@mbarnes This is ready for a first pass review.

/cc @gbraad @intlabs

File: doc/cpd/61.rst
Comment (mbarnes) [inline doc/cpd/61.rst]: nit: it's -> its

File: doc/cpd/61.rst
Comment (mbarnes) [inline doc/cpd/61.rst]: Maybe define "greenfield" here, or link to a definition?  (https://en.wikipedia.org/wiki/Greenfield_project)

File: doc/cpd/61.rst
Comment (mbarnes) [inline doc/cpd/61.rst]: Could we call this "source" instead?  Simpler and, I think, just as clear.

File: doc/cpd/61.rst
Comment (mbarnes) [inline doc/cpd/61.rst]: nit: woul -> would

File: doc/cpd/61.rst
Comment (mbarnes) [inline doc/cpd/61.rst]: nit: excercise -> exercise

File: doc/cpd/61.rst
Comment (mbarnes) [inline doc/cpd/61.rst]: Hmm... could we just say the first `StoreHandler` defined for a particular model type is its default?  I suspect that's going to be most common, and *if* it proves necessary we could add the `default` parameter later as an override.

I'm just thinking about what happens we someone (namely, me) inevitably forgets to tag one of the `StoreHandler` definitions as default.  I think we need a reasonable implicit policy.

**Review by mbarnes — CHANGES_REQUESTED** [review]
Some typos and other thoughts...

File: doc/cpd/61.rst
Comment (ashcrow) [reply doc/cpd/61.rst]: :+1:

File: doc/cpd/61.rst
Comment (ashcrow) [reply doc/cpd/61.rst]: Good idea. Will do.

File: doc/cpd/61.rst
Comment (ashcrow) [reply doc/cpd/61.rst]: Sure.

File: doc/cpd/61.rst
Comment (ashcrow) [reply doc/cpd/61.rst]: That's reasonable. My thought for explicitly stating default was I thought it would be easier to implement but using the first configuration for a model shouldn't be a problem.

**Comment by ashcrow** [comment]
:arrow_up: 

If all looks well please wait until tomorrow morning to merge to give folks outside of US time to review if they would like to do so.

**Review by mbarnes — APPROVED** [review]
Looks great!  I'll wait til tomorrow morning to merge.

**Review by gbraad — APPROVED** [review]
For future: I do not believe 'greenfield' and 'brownfield' not to be explained in the document. I believe you can assume them to be general terms...

File: doc/cpd/61.rst
Comment (gbraad) [inline doc/cpd/61.rst]: Shouldn't this be 'in' ?

Comment (ashcrow) [reply doc/cpd/61.rst]: English is hard. I think you are right. 'in' makes more sense.

**Comment by gbraad** [comment]
Check... besides this it is all fine.

**Review by mbarnes — APPROVED** [review]

**Pull request merged**
